src/providers/WorkflowCore.QueueProviders.SqlServer/Services/SqlServerQueueProvider.cs

```diff
@@ -0,0 +1,157 @@
+﻿#region using
+
+using System;
+using System.Data.SqlClient;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.DependencyInjection;
+
+using WorkflowCore.Interface;
+
+#endregion
+
+namespace WorkflowCore.QueueProviders.SqlServer.Services
+{
+    public class SqlServerQueueProvider : IQueueProvider
+    {
+        readonly string _connectionString;
+
+        readonly bool _canMigrateDb;
+        readonly bool _canCreateDb;
+
+        private readonly IBrokerNamesProvider _names;
+        private readonly ISqlServerQueueProviderMigrator _migrator;
+        private readonly ISqlCommandExecutor _sqlCommandExecutor;
+
+        private readonly string _queueWork;
+        private readonly string _dequeueWork;
+
+        public SqlServerQueueProvider(IServiceProvider serviceProvider, SqlServerQueueProviderOption opt)
+        {
+            _connectionString = opt.ConnectionString;
+            _canMigrateDb = opt.CanMigrateDb;
+            _canCreateDb = opt.CanCreateDb;
+
+            _names = serviceProvider.GetService<IBrokerNamesProvider>()
+                     ?? new BrokerNamesProvider(opt.WorkflowHostName);
+            _sqlCommandExecutor = serviceProvider.GetService<ISqlCommandExecutor>()
```

Comment:
you should be able to just inject the `ISqlCommandExecutor` into the constructor, instead of the service provider

Reply:
Change ctor of SqlServerProvider to remove direct reference to IServiceProvider